task4/src/service/HotelService.java

```diff
@@ -0,0 +1,244 @@
+package service;
+
+import model.Room;
+import model.Service;
+import model.Guest;
+import model.Stay;
+import repository.IHotelRepository;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.ArrayList;
+
+public class HotelService implements IHotelService {
+    private IHotelRepository hotelRepository;
+
+    public HotelService(IHotelRepository hotelRepository) {
+        this.hotelRepository = hotelRepository;
+    }
+
+    @Override
+    public void checkIn(int roomNumber, Guest guest, Date checkInDate, Date checkOutDate) {
+        Room room = hotelRepository.getRoom(roomNumber);
+        if (room != null && room.getStatus().equals("Available")) {
+            room.setStatus("Occupied");
+            Stay stay = new Stay(room, checkInDate, checkOutDate);
+            guest.addStay(stay);
+            hotelRepository.addGuest(guest);
+            System.out.println("Checked in to room: " + roomNumber);
+        } else {
+            System.out.println("Room not available: " + roomNumber);
+        }
+    }
+
+    @Override
+    public void checkOut(int roomNumber) {
+        Room room = hotelRepository.getRoom(roomNumber);
+        if (room != null) {
+            room.setStatus("Available");
+            System.out.println("Checked out from room: " + roomNumber);
+        } else {
+            System.out.println("Room not found: " + roomNumber);
+        }
+    }
+
+    @Override
+    public void setRoomStatus(int roomNumber, String status) {
```

Comment:
```suggestion
    public void setRoomStatus(int roomNumber, String status) {
      Room room = getRoom(roomNumber);
      room.setStatus(status);
      saveRoom(room);
    }
    
    public Room getRoom(int roomNumber) {
      Room room = hotelRepository.getRoom(roomNumber);
      if (room == null) {
        System.out.println("Room not found: " + roomNumber);
        throw . . .
      }
      return room;
    }
```
можно так сделать( и не только здесь но и в остальных похожих методах)